### ISSUE TITLE
Emitting signals from Go defined types

### ISSUE THREAD

**Issue by neclepsio** [go-qml/qml#108]
I cannot find any reference to emitting signals from Go defined types in documentation.

I think the documentation should explain how to make it or report it is not supported.

Anyway, if not supported, it would be cool to add it.

Thanks.

**Comment by pjoe** [comment]
there is a Changed function for signaling property changes, but I don't see a way of generally emitting signals. Would make a lot of sense to add imo

**Comment by niemeyer** [comment]
Indeed, it makes sense to have, and it should be supported at some point. It's just not yet there.
